pkgs/development/perl-modules/generic/default.nix

```diff
@@ -41,8 +41,10 @@ toPerlModule(stdenv.mkDerivation (
     pname = "perl${perl.version}-${lib.getName attrs}"; # TODO: phase-out `attrs.name`
     version = lib.getVersion attrs;                     # TODO: phase-out `attrs.name`
     builder = ./builder.sh;
-    buildInputs = buildInputs ++ [ perl ];
-    nativeBuildInputs = nativeBuildInputs ++ [ (perl.dev or perl) ];
+    # Only add perl.out buildInputs to prevent miniperl (in perl.dev) from being
+    # used by patchShebangs when cross-compiling.
+    buildInputs = buildInputs ++ [ (perl.out or perl) ];
+    nativeBuildInputs = nativeBuildInputs ++ [ (lib.getDev perl) ];
```

Comment:
Please, no.
`perl.dev` is just a wrongly taken name, it has nothing to do with `.dev` outputs of other packages, and must be renamed to `perl.cross` or something.
It is not the first time someone suggest `perl.dev or perl` -> `lib.getDev perl` "optimization" (https://github.com/NixOS/nixpkgs/pull/66458#discussion_r312747456)